drivers/unix/os_unix.cpp

```diff
@@ -126,7 +126,9 @@ void OS_Unix::initialize_core() {
 	RWLockDummy::make_default();
 #else
 	ThreadPosix::make_default();
+#if !defined(OSX_ENABLED)
```

Comment:
Should we also remove it on `IPHONE_ENABLED` and fix #24418?

Reply:
Probably, iOS have its own implementation as well: https://github.com/godotengine/godot/blob/master/platform/iphone/os_iphone.cpp#L99, but I have not tested it.